src/components/DataCollector/index.js

```diff
@@ -37,8 +38,17 @@ const createDataCollector = ({ eventManager, logger }) => {
             event.documentMayUnload();
           }
 
-          event.setUserXdm(xdm);
-          event.setUserData(data);
+          if (xdm) {
+            // Clone xdm in case the customer modifies the object
+            // between now and the time it gets sent to the server.
+            event.setUserXdm(clone(xdm));
```

Comment:
Like the comment says, DataCollector is now cloning `xdm` and `data` as soon as possible so that if the customer modifies the object between now and the time it gets sent to the server, we use the original state and not the state after modification. What's interesting is that the command option validator we're using (which leverages Yup) _also_ clones this data in some fashion, making this particular `clone` call unnecessary to accomplish our goal, but yet I prefer that we explicitly clone here anyway so that our intentions are explicit and we're not relying on the validator to do the cloning.